examples/vision/fully_convolutional_network.py

```diff
@@ -0,0 +1,651 @@
+"""
+Title: Image Segmentation using Composable Fully-Convolutional Networks
+Author: [Suvaditya Mukherjee](https://twitter.com/halcyonrayes)
+Date created: 2023/05/08
+Last modified: 2023/05/08
+Description: Using the Fully-Convolutional Network for Image Segmentation.
+Accelerator: GPU
+"""
+
+"""
+## Introduction
+
+The following example walks through the steps to implement Fully-Convolutional Networks
+for Image Segmentation on the Oxford-IIIT Pets dataset. The model was proposed in the
+paper, [Fully Convolutional Networks for Semantic Segmentation by Long et. al.
+(2014)](https://arxiv.org/abs/1411.4038).
+Image segmentation is one of the most common and introductory tasks when it comes to
+Computer Vision. The idea is to extend the problem of Image Classification from
+one-label-per-image to a pixel-wise classification problem.
+In this example, we are going to work on developing a Fully-Convolutional Network that is
+capable of performing Image Segmentation. For this, we make use of the VGG-19 model as
+available in `keras.applications` and continue towards extending its internal layers and
+creating our Fully-Convolutional Network.
+The network extends the pooling layer outputs from the VGG in order to perform upsampling
+and get a final result. Different intermediate pool layers are extracted and processed
+upon for different versions of the network.
+The FCN architecture has 3 versions of differing quality, as mentioned in the paper.
+- FCN-32S
+- FCN-16S
+- FCN-8S
+All versions of the model derive their outputs through an iterative processing of
+successive intermediate pool layers of the main backbone used. A better idea can be
+gained from the figure below.
+| ![FCN Architecture](https://i.imgur.com/Ttros06.png) |
+| :--: |
+| **Diagram 1**: Combined Architecture Versions (Source: Paper)|
+
+The process we are going to follow is:
+- Set-up all imports
```

Comment:
Maybe just let the section headers showcase this - no need to provide an outline for a short example.

Reply:
Got it, removed this part.